### PR TITLE
💚 Fix r.join is not a function

### PR DESCRIPTION
**Pull request by JackDevAU**
- Attempts to fix - #5226

Tested Build, 

- ✅ Locally 
- ✅ GitHub Pages
- ✅ Self hosted
- ✅ Vercel

### CODE REVIEW

File: .changeset/good-buses-mix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"tinacms": patch
+---
+
+⚠️  Don't Merge ⚠️  makeCacheDir Check
```
Comment (JackDevAU) [inline .changeset/good-buses-mix.md:5]:
```suggestion
---
"tinacms": patch
"@tinacms/cli": patch
---

⚠️  Don't Merge ⚠️  makeCacheDir Check
```

File: .changeset/good-buses-mix.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"tinacms": patch
+"@tinacms/cli": patch
+---
+
+⚠️  Don't Merge ⚠️  makeCacheDir Check
```
Comment (JackDevAU) [inline .changeset/good-buses-mix.md:6]:
```suggestion
🐛 fix r.join error when building pages
```

File: packages/tinacms/src/cache/node-cache.ts [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
   }
 
-  fs.mkdirSync(cacheDir, { recursive: true })
   return cacheDir
 }
 
 export const NodeCache = async (dir: string): Promise<Cache> => {
-  const fs = await import('node:fs')
-  const { createHash } = await import('node:crypto')
+  // Change to require?
```
Comment (JackDevAU) [inline packages/tinacms/src/cache/node-cache.ts:35]:
```suggestion

```

File: packages/tinacms/src/cache/node-cache.ts [hunk 4]
```diff
@@ -1,25 +1,40 @@
...
 
+  const pathParts = dir.split(path.sep)
+  const cacheHash = pathParts[pathParts.length - 1]
   let cacheDir = dir
-  // check if the root directory exists, if not then create create in tmp and return new path
-  if (!fs.existsSync(path.join(path.sep, parts[0]))) {
-    cacheDir = path.join(os.tmpdir(), parts[parts.length - 1])
+
+  // ["Users", "jackpettit", "Developer", "tina", "tina-cloud-starter", "tina", "__generated__", "".cache", "1729659325506"]
+  // "/tmp/17.../"
```
Comment (ncn-ssw) [inline packages/tinacms/src/cache/node-cache.ts:19]: **question**: What is the purpose of this comment?

**Comment by ncn-ssw** [comment]
**question**: Given that we've had several issues with the caching system, can we automate the tests?

@JackDevAU

File: packages/tinacms/src/cache/node-cache.ts [hunk 5]
```diff
@@ -1,25 +1,40 @@
...
+
+  if (!fs.existsSync(dir)) {
```
Comment (brookjeynes-ssw) [inline packages/tinacms/src/cache/node-cache.ts:21]: This won't work as `dir` won't exist yet. We should check if `.../.cache` exists and if it doesn't, resort to `/tmp`

Comment (ncn-ssw) [reply packages/tinacms/src/cache/node-cache.ts:21]: `dir` is a parameter to the function (line 5 after changes), and the existence of a valid value is checked on line 10 (after changes).

Comment (brookjeynes-ssw) [reply packages/tinacms/src/cache/node-cache.ts:21]: Sorry @ncn-ssw, I should have been more clear. The system directory that `dir` references won't exist therefor `existsSync` will always result in false

File: packages/@tinacms/cli/src/next/vite/index.ts [hunk 6]
```diff
@@ -155,17 +155,22 @@ export const createConfig = async ({
...
-    path: path.join(configManager.spaRootPath, 'src', 'dummy-client.ts'),
+    ...(isBrowser && {
+      crypto: path.join(configManager.spaRootPath, 'src', 'dummy-client.ts'),
+      fs: path.join(configManager.spaRootPath, 'src', 'dummy-client.ts'),
+      os: path.join(configManager.spaRootPath, 'src', 'dummy-client.ts'),
+      path: path.join(configManager.spaRootPath, 'src', 'dummy-client.ts'),
+    }),
   }
+
   if (configManager.shouldSkipSDK()) {
```
Comment (kldavis4) [inline packages/@tinacms/cli/src/next/vite/index.ts:174]: why are these changes needed? Isn't this only executed in the cli dev/build commands?

Comment (JackDevAU) [reply packages/@tinacms/cli/src/next/vite/index.ts:174]: I originally thought these could potentially be causing the issue as to why the import wasn't working. After testing this some more - I have noticed that isBrowser is always false. So Will remove these 🙏

**Review by kldavis4 — APPROVED** [review]

**Review by kldavis4 — APPROVED** [review]

**Review by brookjeynes-ssw — APPROVED** [review]
